inst/sql/modified_update/migrate_coverage.sql

```diff
@@ -5,9 +5,10 @@ coverage_old.id AS coverage_set_id_old,
 coverage_new.id AS coverage_set_id_new
 FROM (SELECT * FROM coverage_set
       WHERE touchstone != '{{{touchstone_new}}}') AS coverage_old
-LEFT JOIN (SELECT coverage_set.* FROM coverage_set
+RIGHT JOIN (SELECT coverage_set.* FROM coverage_set
```

Comment:
can you try a `FULL OUTER JOIN` here instead and see if that makes sense for the cases that require both?  (see here: https://stackoverflow.com/a/16598900).  If that works that'll be the most simple.  

Otherwise can you modify the `modified_update_calculate` function to take a new logical argument with a descriptive name and document it in the format as above the function (i.e. with @param).  Set it with a default that corresponds to the case of modup-201710 (please use a different name than right/left and one that refers to the underlying problem which I'm afraid I've not managed to understand yet).  Pass that argument through to `mu_prepare` and then let me know - I'll patch the sql query so we can switch between the two forms

Reply:
I suppose it should be INNER JOIN, instead of FULL OUTER JOIN. 

But, to me, the best solution is: keep LEFT join, so that any old impact will join the process of modified update; when there is 'no new coverage set found', instead of terminating the modup process with an error message, we use coverage_old as coverage_new (meaning not updating) following a warning message for the reporter. What do you think?

Reply:
Sorry, I don't understand this, and I think it's something that needs sorting out ahead of time rather than trying various combinations until the right things happens

Reply:
This is the bit that needs reverting if you want the other changes merged straight away.

Reply:
OK. I will revert is back to LEFT JOIN.

Reply:
Reverted.